streampark-console/streampark-console-service/src/test/java/org/apache/org/apache/streampark/common/util/CompletableFutureUtilsTest.java

```diff
@@ -143,4 +147,77 @@ private String runStart(int sec) {
     }
     return "start successful";
   }
+
+  @Test
+  public void thenSupplyNormally() throws Exception {
+    String successResult = "success";
+    String exceptionResult = "error";
+
+    String resp =
+        CompletableFutureUtils.supplyTimeout(
+                CompletableFuture.supplyAsync(() -> successResult),
+                3,
+                TimeUnit.SECONDS,
+                success -> success,
+                e -> exceptionResult)
+            .thenApply(r -> r)
+            .get();
+
+    Assertions.assertEquals(resp, successResult);
+  }
+
+  @Test
+  public void thenSupplyTimeout() throws Exception {
+    String successResult = "success";
+    String exceptionResult = "error";
+    CompletableFuture<String> future =
+        CompletableFuture.supplyAsync(
+            () -> {
+              try {
+                Thread.sleep(5000);
+              } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+              }
+              return successResult;
+            });
+    String resp =
+        CompletableFutureUtils.supplyTimeout(
+                future, 1, TimeUnit.SECONDS, success -> success, e -> exceptionResult)
+            .thenApply(r -> r)
+            .get();
+    Assertions.assertEquals(resp, exceptionResult);
+  }
+
+  @Test
+  public void thenSupplyException() {
+    String resp;
+    String exResult = "exception";
+    try {
+      resp = exceptionally();
+    } catch (Exception e) {
+      resp = exResult;
+    }
+    Assertions.assertEquals(resp, exResult);
+  }
+
+  private String exceptionally() throws Exception {
+    return CompletableFutureUtils.supplyTimeout(
+            CompletableFuture.supplyAsync(
+                () -> {
+                  try {
+                    Thread.sleep(5000);
+                  } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                  }
+                  return "success";
+                }),
+            2,
+            TimeUnit.SECONDS,
+            success -> success,
+            e -> {
+              throw new RuntimeException("exception");
+            })
+        .thenApply(r -> r)
+        .get();
+  }
```

Comment:
```suggestion

  @Test
  public void thenSupplyException() {
    String expectedExceptionMessage = "Exception in exceptionally handler";
    CompletableFuture<String> future =
        CompletableFutureUtils.supplyTimeout(
            CompletableFuture.supplyAsync(
                () -> {
                  try {
                    Thread.sleep(5000);
                  } catch (InterruptedException e) {
                    throw new RuntimeException(e);
                  }
                  return "success";
                }),
            2,
            TimeUnit.SECONDS,
            success -> success,
            e -> {
              throw new RuntimeException(expectedExceptionMessage);
            });
    assertThatThrownBy(future::get)
        .as(expectedExceptionMessage)
        .isInstanceOf(RuntimeException.class);
  }
```

Using the `assertThatThrownBy` instead of try catch.

Reply:
We can catch this exception instead of actually throw this exception, otherwise the test ci verification failed will fail

Reply:
You can take a look how to use the  `assertThatThrownBy`. Apache flink use it as well.